### PR TITLE
mount user's volume

### PR DESCRIPTION
**Pull request by Tianhao-Gu**

### CODE REVIEW

File: src/jupyterhub_config/custom_docker_spawner.py [hunk 1]
```diff
@@ -131,3 +131,39 @@ def _configure_notebook_dir(self, username: str, user_dir: Path):
...
+        return self.user.admin or self.RW_MINIO_GROUP in group_names
+
+    def _ensure_user_volume(self):
+        """
+        Ensure the user's volume is correctly mounted in the container.
+        """
+
+        user_home_dir = Path(os.environ['JUPYTERHUB_USER_HOME'])
+        mount_base_dir = Path(os.environ['JUPYTERHUB_MOUNT_BASE_DIR'])
+        hub_secrets_dir = Path(os.environ['JUPYTERHUB_SECRETS_DIR'])
```
Comment (MrCreosote) [inline src/jupyterhub_config/custom_docker_spawner.py:151]: Remind me what's in the secrets dir, I've already forgotten

File: src/jupyterhub_config/custom_docker_spawner.py [hunk 2]
```diff
@@ -131,3 +131,39 @@ def _configure_notebook_dir(self, username: str, user_dir: Path):
...
+
+        cdm_shared_dir = Path(os.environ['CDM_SHARED_DIR'])  # Legacy data volume from JupyterLab
```
[the remark this answers is not in the document]
Comment (Tianhao-Gu) [reply src/jupyterhub_config/custom_docker_spawner.py:153]: hmmm I am thinking we will need to mount a shared dir for people to put shared notebooks. Also, I am planning to use admin account to move existing notebooks to user's new workspace. For the later, there are other ways.

File: src/jupyterhub_config/custom_docker_spawner.py [hunk 3]
```diff
@@ -131,3 +131,39 @@ def _configure_notebook_dir(self, username: str, user_dir: Path):
...
+        hive_metastore_dir = Path(os.environ['HIVE_METASTORE_DIR'])  # within cdm_shared_dir
+
+        if self.user.admin:
+            self.log.info(f'Admin user detected: {self.user.name}. Setting up admin mount points.')
+            self.volumes.update({
+                f'{mount_base_dir}/{user_home_dir}': f'{user_home_dir}',  # Global users home directory
+                f'{mount_base_dir}/{hub_secrets_dir}': f'{hub_secrets_dir}',
+                f'{mount_base_dir}/{cdm_shared_dir}': f'{cdm_shared_dir}',  # Legacy data volume from JupyterLab
```
[the remark this answers is not in the document]
Comment (Tianhao-Gu) [reply src/jupyterhub_config/custom_docker_spawner.py:161]: hive is within `cdm_shared_dir`. So I didn't mount it here specifically.

File: src/jupyterhub_config/custom_docker_spawner.py (hunk 1 above)
Comment (Tianhao-Gu) [reply src/jupyterhub_config/custom_docker_spawner.py:151]: it holds cookie sessions and user DB.

File: src/jupyterhub_config/custom_docker_spawner.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (MrCreosote) [reply src/jupyterhub_config/custom_docker_spawner.py:153]: So it's not really legacy then?

File: src/jupyterhub_config/custom_docker_spawner.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (MrCreosote) [reply src/jupyterhub_config/custom_docker_spawner.py:161]: But didn't we say it's not needed in a recent pr?

File: src/jupyterhub_config/custom_docker_spawner.py (hunk 1 above)
Comment (MrCreosote) [reply src/jupyterhub_config/custom_docker_spawner.py:151]: :+1:

File: src/jupyterhub_config/custom_docker_spawner.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (Tianhao-Gu) [reply src/jupyterhub_config/custom_docker_spawner.py:153]: hmmm yea. Maybe I will remove the comments or remove the mount once we are set stone of mounting volume.

File: src/jupyterhub_config/custom_docker_spawner.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (Tianhao-Gu) [reply src/jupyterhub_config/custom_docker_spawner.py:161]: I think you are referring to postgres dir which is remote now so we don't need it. I think users need to read hive in order to read correctly.

[the remark this answers is not in the document]
Comment (Tianhao-Gu) [reply src/jupyterhub_config/custom_docker_spawner.py:161]: seems like when a namespace is created in postgres, an associated namespace.db file is created in the hive dir. So I am guessing it's needed. But we can remove it if we noticed it's no longer needed. I want to grant users broad privileges initially, then scale them back gradually based on our findings. That way we can move faster.

[the remark this answers is not in the document]
Comment (MrCreosote) [reply src/jupyterhub_config/custom_docker_spawner.py:161]: :+1:

File: src/jupyterhub_config/custom_docker_spawner.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (MrCreosote) [reply src/jupyterhub_config/custom_docker_spawner.py:153]: :+1:

**Review by MrCreosote — APPROVED** [review]

**Pull request merged**